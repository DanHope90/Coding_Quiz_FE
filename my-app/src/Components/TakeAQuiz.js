/* eslint-disable no-console */
import React, { useState, useEffect } from "react";
import axios from "axios";
import PropTypes from "prop-types";
import TakeAQuizCard from "./TakeAQuizCard";
import "../Styles/TakeAQuiz.css";

function TakeAQuiz(quizId) {
  // Below sets the initial alert.
  const initialState = {
    alert: { message: "", isSuccess: false },
  };

  const [alert, setAlert] = useState(initialState.alert);

  // Below employs an empty array to store all quizzes, useEffect makes an axios request
  // to populate said array with all quizzes. This may not be needed later.
  const [availableQuizzes, setAvailableQuizzes] = useState([]);

  useEffect(() => {
    axios
      .get("http://localhost:4000/api/quizzes")
      .then(({ data }) => setAvailableQuizzes(data))
      .catch(() => setAlert({ message: "Axios promise rejected", isSuccess: false }));
  }, []);

  console.log(availableQuizzes);
  console.log(initialState);

  // Below employs an empty object to store our desired quiz.
  // useEffect makes an axios request to populate object with
  // the quiz we have chosen, using quizId as a prop passed down.

  const [desiredQuiz, setDesiredQuiz] = useState({});

  useEffect(() => {
    axios
      .get(`http://localhost:4000/api/quizzes/${quizId}`)
      .then(({ data }) => setDesiredQuiz(data))
      .catch(() => setAlert({ message: "Axios promise rejected", isSuccess: false }));
  });

  console.log(desiredQuiz);

  return (
    <>
      <div>This is the Take A Quiz Page</div>
      <div className="quiz-array">
        {availableQuizzes.map((quiz) => (
          <div key={quiz.quizName} className="indiv.quiz">
            <TakeAQuizCard quizName={quiz.quizName} quizURL={quiz.quizURL} />
          </div>
        ))}
      </div>
    </>
  );
}

export default TakeAQuiz;

TakeAQuiz.propTypes = {
  quizName: PropTypes.string,
  quizURL: PropTypes.string,
}.isRequired;
