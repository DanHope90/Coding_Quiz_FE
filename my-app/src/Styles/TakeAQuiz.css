.take-quiz-array{
    display: flex;
    flex-flow: wrap;
    justify-content: center;
    text-align: center;
}

.take-indiv-quiz{
    padding: 5px;
    margin: 5px;
}

.leave-quiz-button{
    color: red;
    width: 90%;
    height: 50px;
}

.leave-quiz-button:hover{
    background-color: red;
    color: black;
}

.leave-quiz-button:active{
    background-color: white;
    color: red;
}