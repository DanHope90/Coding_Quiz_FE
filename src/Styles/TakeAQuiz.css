.select-a-quiz-text {
    font-size: inherit;
}

.execute-quiz {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    gap: 20px;
}

.take-quiz-array{
    display: flex;
    flex-flow: wrap;
    justify-content: center;
    text-align: center;
}

.take-indiv-quiz{
    padding: 5px;
    margin: 5px;
}

.leave-quiz-button{
    color: red;
    width: 90%;
    height: 50px;
}

.leave-quiz-button:hover{
    background-color: red;
    color: black;
}

.leave-quiz-button:active{
    background-color: white;
    color: red;
}


@media only screen and (max-width: 650px) {
    .select-a-quiz-text {
        font-size: 12px;
    }
}