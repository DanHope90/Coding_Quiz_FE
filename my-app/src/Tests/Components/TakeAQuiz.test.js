import React from "react";
import renderer from "react-test-renderer";
import TakeAQuiz from "../../Components/TakeAQuiz";

test("renders the TakeAQuiz component correctly", () => {
  const testRenderer = renderer.create(<TakeAQuiz />);
  expect(testRenderer).toMatchSnapshot();
});
