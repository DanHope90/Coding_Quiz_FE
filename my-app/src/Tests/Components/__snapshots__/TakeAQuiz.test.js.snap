// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TakeAQuiz component correctly 1`] = `
Array [
  <div>
    This is the Take A Quiz Page
  </div>,
  <div
    className="quiz-array"
  />,
]
`;
